### ISSUE TITLE
NDK stacktrace is not detailed enough

### ISSUE THREAD

**Issue by peterdk** [bugsnag/bugsnag-android#380]
### Expected behavior
I get NDK stacktraces from crashes within the JNI code, but unfortunately it doesnt show the correct line of the crash. It only shows the linenumber where the method itself ends, and doesnt show where in the method something goes wrong.

I would expect to pinpoint the exact line where the crash happens

### Observed behavior
It shows the line where the method itself ends, not the crash location exactly. Also it says 'unknown symbol'.

### Steps to reproduce
I added a test nullpointer crash inside the method

### Version
3.8.2
armeabi-v7a 
Gradle uploads mapping and all .so file mappings correctly.

### Additional information
```
SIGILL: Illegal instruction
        at bsg_unwind_stack_libunwind(/data/app/FOO/lib/arm/libbugsnag-ndk.so:27562)
        at bsg_unwind_stack(/data/app/FOO/lib/arm/libbugsnag-ndk.so:26862)
        at bsg_handle_signal(/data/app/FOO/lib/arm/libbugsnag-ndk.so:26316)
        at 0xf4c6996c(/system/lib/libc.so:100716)
        at unknown_symbol(C:/Users/Peter/git/FOO/src/main/jni/FileCache.cpp:62)
        at unknown_symbol(C:/Users/Peter/git/FOO/src/main/jni/SamplerSynth.cpp:32)
```
```
MemoryStream * FileCache::getBackedStream(int note)
{
	if (note < 0 || note >= cacheSize)
	{
		ALOGD("Note %d is larger then stream cache size %d", cacheSize);
		return NULL;
	}

	MemoryStream * test = NULL;
	__umito_log_print(LOG_INFO, "FileCache","%d", test->getBytesOutputted()); //CRASH
    if (backedStreamCaches != NULL && backedStreamCaches[note] != 0)
    {
        int streamLength = backedStreamLengths[note];
		return new MemoryStream(backedStreamCaches[note], streamLength, false);
    }else{
        __umito_log_print(LOG_INFO, "FileCache", "Returning NULL stream for note %d: no cached stream available", note);
        return NULL;
    }
}//Line 62 is this line
```
I also tried compiling without -O2, and with -fno-omit-frame-pointer -funwind-tables -Wl,--no-merge-exidx-entries. No different results there.

**Comment by peterdk** [comment]
Same issue on x86.

**Comment by kattrali** [comment]
Thanks for the report, @peterdk. Looking into it, there should be some improvements here shipping soon.

**Comment by peterdk** [comment]
@kattrali Interesting, I am indeed restricting my NDK libraries to be armeabi-v7a and x86. I don't ship ARM64 libs.

**Comment by kattrali** [comment]
> But the single line crash is also present on devices that are pure armeabi-v7a. It doesn't only happen for ARM64 devices.

This part should be resolved in 4.9.2

**Comment by fractalwrench** [comment]
We believe recent fixes to the bugsnag-android-ndk library and on our backend should resolve this issue, so I'm closing off this ticket for now.
